internal/controllers/configuration/kongupstreampolicy_utils.go

```diff
@@ -198,9 +262,67 @@ func getAllBackendRefsUsedWithService(httpRoute gatewayapi.HTTPRoute, serviceKey
 	return backendRefs
 }
 
-// -----------------------------------------------------------------------------
-// KongUpstreamPolicy Controller - Helpers
-// -----------------------------------------------------------------------------
+func (r *KongUpstreamPolicyReconciler) buildPolicyStatus(
+	upstreamPolicyNN k8stypes.NamespacedName,
+	ancestorsStatus []ancestorStatus,
+) (gatewayapi.PolicyStatus, error) {
+	// Sort the ancestors by creation timestamp and keep only the oldest ones.
+	sort.Slice(ancestorsStatus, func(i, j int) bool {
+		return ancestorsStatus[i].creationTimestamp.Before(&ancestorsStatus[j].creationTimestamp)
+	})
+	if len(ancestorsStatus) > maxNAncestors {
```

Comment:
`maxNAncestors` feels like an implementation detail that users should be aware of. We're not documenting this but this is being enforced on the Gateway API level in `gatewayv1alpha2.PolicyStatus`. Perhaps it's worth documenting it in a comment and in the log message?